src/main/java/kitchenpos/menu/application/MenuService.java

```diff
@@ -0,0 +1,52 @@
+package kitchenpos.menu.application;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import kitchenpos.menu.application.dto.MenuCreateRequest;
+import kitchenpos.menu.application.dto.MenuResponse;
+import kitchenpos.menu.application.dto.MenusResponse;
+import kitchenpos.menu.domain.Menu;
+import kitchenpos.menu.domain.MenuProduct;
+import kitchenpos.menu.domain.MenuRepository;
+import kitchenpos.menu.domain.MenuValidator;
+import kitchenpos.menugroup.domain.MenuGroup;
+import kitchenpos.menugroup.domain.MenuGroupRepository;
+import kitchenpos.menugroup.exception.MenuGroupException.NotFoundMenuGroupException;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+public class MenuService {
+    private final MenuRepository menuRepository;
+    private final MenuGroupRepository menuGroupRepository;
+    private final MenuValidator menuValidator;
+
+    public MenuService(
+            final MenuRepository menuRepository, final MenuGroupRepository menuGroupRepository,
+            final MenuValidator menuValidator) {
+        this.menuRepository = menuRepository;
+        this.menuGroupRepository = menuGroupRepository;
+        this.menuValidator = menuValidator;
+    }
+
+    @Transactional
+    public MenuResponse create(final MenuCreateRequest request) {
+        final MenuGroup menuGroup = menuGroupRepository.findById(request.getMenuGroupId())
+                .orElseThrow(NotFoundMenuGroupException::new);
+
+        List<MenuProduct> menuProducts = request.getMenuProducts().stream()
+                .map(menuProductRequest -> new MenuProduct(
+                        menuProductRequest.getProductId(), menuProductRequest.getQuantity())
+                ).collect(Collectors.toUnmodifiableList());
+
+        menuValidator.validateCreate(request.getPrice(), menuProducts);
+        final Menu menu = new Menu(request.getName(), request.getPrice(), menuGroup, menuProducts);
```

Comment:
```suggestion
        final Menu menu = new Menu(request.getName(), request.getPrice(), menuGroup, menuProducts);
        menu.register(menuValidator);
```
위와 같은 방식은 어떨까요?
Service Layer에서 validator를 직접적으로 호출한 뒤 생성하는 것도 문제는 없으나, 메뉴 등록에 대한 행위를 명확히 나타낸다는 점에서 위 방식도 괜찮다고 생각이 드네요~

의견일 뿐 반영하지 않으셔도 괜찮고 성하의 생각만 말씀해 주시면 감사하겠습니다 :)

Reply:
오! 이렇게 할 수도 있군요!!! 다양한 방법 알려주셔서 감사합니다 ㅎㅎㅎ

말랑이 Suggestion 해주신 코드에서는 Menu를 생성한 후에 MenuValidator로 검증을 수행하는데요!
저는 MenuValidator에서 Create 검증을 수행한 후에 Menu가 생성되는게 좋다고 생각해서
기존 코드를 바꾸지 않으려고 하는데 괜찮을까요?!

Reply:
넹 괜찮습니다~